guides/common/modules/con_understanding-puppet-architecture.adoc

```diff
@@ -0,0 +1,19 @@
+[id="Understanding_puppet_architecture_{context}"]
+= Understanding Puppet Architecture
+
+Puppet uses client-server architecture with managed hosts running the Puppet agent and {Project} or any {SmartProxies} acting as a Puppet server.
+
+The Puppet agent collects _facts_ about a host and reports them to the Puppet server on each run.
+The Puppet server compares the reported _facts_ with the system state defined for the host.
```

Comment:
Puppet server actually forwards them to the {Project} and the {Project} stores them for later use. Based on the _facts_ and other definitions, it is able to construct the ENC answer to the Puppet server.